File: backend-java/src/main/java/mx/codemx/modules/duelos/DueloRepository.java
package mx.codemx.modules.duelos;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface DueloRepository extends JpaRepository<Duelo, Long> {

    @Query("select d from Duelo d where d.jugador1Id = :usuarioId or d.jugador2Id = :usuarioId "
            + "order by d.creadoEn desc")
    List<Duelo> listarPorUsuario(@Param("usuarioId") long usuarioId);
}

File: backend/Modules/Duelos/DueloRepository.cs
using CodeMX.Api.Persistence;
using Microsoft.EntityFrameworkCore;

namespace CodeMX.Api.Modules.Duelos;

/// <summary>Acceso a datos del módulo Duelos.</summary>
public class DueloRepository
{
    private readonly CodeMxDbContext _db;

    public DueloRepository(CodeMxDbContext db) => _db = db;

    public async Task<Duelo> Guardar(Duelo duelo)
    {
        if (duelo.Id == 0) _db.Duelos.Add(duelo);
        else _db.Duelos.Update(duelo);
        await _db.SaveChangesAsync();
        return duelo;
    }

    public Task<Duelo?> BuscarPorId(long id) => _db.Duelos.FirstOrDefaultAsync(d => d.Id == id);

    public Task<List<Duelo>> ListarPorUsuario(long usuarioId) =>
        _db.Duelos
           .Where(d => d.Jugador1Id == usuarioId || d.Jugador2Id == usuarioId)
           .OrderByDescending(d => d.CreadoEn)
           .ToListAsync();
}
